0    [main] INFO  kafka.utils.VerifiableProperties  - Verifying properties
27   [main] WARN  kafka.utils.VerifiableProperties  - Property auto.offset.reset is not valid
27   [main] WARN  kafka.utils.VerifiableProperties  - Property destination.topic is not valid
28   [main] WARN  kafka.utils.VerifiableProperties  - Property group.id is not valid
28   [main] INFO  kafka.utils.VerifiableProperties  - Property metadata.broker.list is overridden to localhost:9092
28   [main] WARN  kafka.utils.VerifiableProperties  - Property source.topic is not valid
28   [main] WARN  kafka.utils.VerifiableProperties  - Property zookeeper.connect is not valid
73   [main] INFO  kafka.utils.VerifiableProperties  - Verifying properties
73   [main] INFO  kafka.utils.VerifiableProperties  - Property auto.offset.reset is overridden to largest
73   [main] WARN  kafka.utils.VerifiableProperties  - Property destination.topic is not valid
73   [main] INFO  kafka.utils.VerifiableProperties  - Property group.id is overridden to space.exploration.mars.rover
73   [main] WARN  kafka.utils.VerifiableProperties  - Property metadata.broker.list is not valid
73   [main] WARN  kafka.utils.VerifiableProperties  - Property source.topic is not valid
73   [main] INFO  kafka.utils.VerifiableProperties  - Property zookeeper.connect is overridden to localhost:2181
5107 [main] INFO  kafka.consumer.ZookeeperConsumerConnector  - [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b], Connecting to zookeeper instance at localhost:2181
5116 [ZkClient-EventThread-15-localhost:2181] INFO  org.I0Itec.zkclient.ZkEventThread  - Starting ZkClient event thread.
10126 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:zookeeper.version=3.4.6-1569965, built on 02/20/2014 09:09 GMT
10126 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:host.name=10.0.0.229
10126 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:java.version=1.8.0_121
10126 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:java.vendor=Oracle Corporation
10126 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:java.home=/Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home/jre
10126 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:java.class.path=/Users/sanketkorgaonkar/Documents/workspace/mars.rover.project/mars.rover/target/classes:/Users/sanketkorgaonkar/.m2/repository/com/google/protobuf/protobuf-java/3.0.0/protobuf-java-3.0.0.jar:/Users/sanketkorgaonkar/.m2/repository/org/apache/kafka/kafka_2.10/0.9.0.1/kafka_2.10-0.9.0.1.jar:/Users/sanketkorgaonkar/.m2/repository/com/101tec/zkclient/0.7/zkclient-0.7.jar:/Users/sanketkorgaonkar/.m2/repository/org/slf4j/slf4j-api/1.6.1/slf4j-api-1.6.1.jar:/Users/sanketkorgaonkar/.m2/repository/log4j/log4j/1.2.15/log4j-1.2.15.jar:/Users/sanketkorgaonkar/.m2/repository/javax/mail/mail/1.4/mail-1.4.jar:/Users/sanketkorgaonkar/.m2/repository/javax/activation/activation/1.1/activation-1.1.jar:/Users/sanketkorgaonkar/.m2/repository/org/apache/kafka/kafka-clients/0.9.0.1/kafka-clients-0.9.0.1.jar:/Users/sanketkorgaonkar/.m2/repository/org/xerial/snappy/snappy-java/1.1.1.7/snappy-java-1.1.1.7.jar:/Users/sanketkorgaonkar/.m2/repository/net/jpountz/lz4/lz4/1.2.0/lz4-1.2.0.jar:/Users/sanketkorgaonkar/.m2/repository/com/yammer/metrics/metrics-core/2.2.0/metrics-core-2.2.0.jar:/Users/sanketkorgaonkar/.m2/repository/org/scala-lang/scala-library/2.10.5/scala-library-2.10.5.jar:/Users/sanketkorgaonkar/.m2/repository/net/sf/jopt-simple/jopt-simple/3.2/jopt-simple-3.2.jar:/Users/sanketkorgaonkar/.m2/repository/org/slf4j/slf4j-log4j12/1.7.6/slf4j-log4j12-1.7.6.jar:/Users/sanketkorgaonkar/.m2/repository/org/apache/zookeeper/zookeeper/3.4.6/zookeeper-3.4.6.jar:/Users/sanketkorgaonkar/.m2/repository/jline/jline/0.9.94/jline-0.9.94.jar:/Users/sanketkorgaonkar/.m2/repository/io/netty/netty/3.7.0.Final/netty-3.7.0.Final.jar
10127 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:java.library.path=/Users/sanketkorgaonkar/Library/Java/Extensions:/Library/Java/Extensions:/Network/Library/Java/Extensions:/System/Library/Java/Extensions:/usr/lib/java:.
10127 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:java.io.tmpdir=/var/folders/5h/gwr26fh90zs2zgrhjqr0pm1r0000gn/T/
10127 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:java.compiler=<NA>
10127 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:os.name=Mac OS X
10127 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:os.arch=x86_64
10127 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:os.version=10.12.4
10127 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:user.name=sanketkorgaonkar
10127 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:user.home=/Users/sanketkorgaonkar
10127 [main] INFO  org.apache.zookeeper.ZooKeeper  - Client environment:user.dir=/Users/sanketkorgaonkar/Documents/workspace/mars.rover.project/mars.rover
10128 [main] INFO  org.apache.zookeeper.ZooKeeper  - Initiating client connection, connectString=localhost:2181 sessionTimeout=6000 watcher=org.I0Itec.zkclient.ZkClient@212bf671
10143 [main] INFO  org.I0Itec.zkclient.ZkClient  - Waiting for keeper state SyncConnected
10146 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error)
10202 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Socket connection established to localhost/127.0.0.1:2181, initiating session
10210 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Session establishment complete on server localhost/127.0.0.1:2181, sessionid = 0x15b9b2525a100d4, negotiated timeout = 6000
10211 [main-EventThread] INFO  org.I0Itec.zkclient.ZkClient  - zookeeper state changed (SyncConnected)
10219 [main] INFO  kafka.consumer.ZookeeperConsumerConnector  - [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b], starting auto committer every 60000 ms
10241 [Thread-2] INFO  kafka.consumer.ZookeeperConsumerConnector  - [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b], begin registering consumer space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b in ZK
10281 [Thread-2] INFO  kafka.utils.ZKCheckedEphemeral  - Creating /consumers/space.exploration.mars.rover/ids/space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b (is it secure? false)
10370 [Thread-2] INFO  kafka.utils.ZKCheckedEphemeral  - Result of znode creation is: OK
10370 [Thread-2] INFO  kafka.consumer.ZookeeperConsumerConnector  - [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b], end registering consumer space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b in ZK
10389 [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b_watcher_executor] INFO  kafka.consumer.ZookeeperConsumerConnector  - [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b], starting watcher executor thread for consumer space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b
10423 [Thread-2] INFO  kafka.consumer.ZookeeperConsumerConnector  - [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b], begin rebalancing consumer space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b try #0
10558 [Thread-2] INFO  kafka.consumer.ConsumerFetcherManager  - [ConsumerFetcherManager-1493604875552] Stopping leader finder thread
10558 [Thread-2] INFO  kafka.consumer.ConsumerFetcherManager  - [ConsumerFetcherManager-1493604875552] Stopping all fetchers
10559 [Thread-2] INFO  kafka.consumer.ConsumerFetcherManager  - [ConsumerFetcherManager-1493604875552] All connections stopped
10560 [Thread-2] INFO  kafka.consumer.ZookeeperConsumerConnector  - [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b], Cleared all relevant queues for this fetcher
10561 [Thread-2] INFO  kafka.consumer.ZookeeperConsumerConnector  - [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b], Cleared the data chunks in all the consumer message iterators
10561 [Thread-2] INFO  kafka.consumer.ZookeeperConsumerConnector  - [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b], Committing all offsets after clearing the fetcher queues
10562 [Thread-2] INFO  kafka.consumer.ZookeeperConsumerConnector  - [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b], Releasing partition ownership
10607 [Thread-2] INFO  kafka.consumer.RangeAssignor  - Consumer space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b rebalancing the following partitions: ArrayBuffer() for topic secure_com_from_earth_channel_1 with consumers: List(space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604858171-361cb25e-0, space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b-0)
10610 [Thread-2] WARN  kafka.consumer.RangeAssignor  - No broker partitions consumed by consumer thread space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604858171-361cb25e-0 for topic secure_com_from_earth_channel_1
10610 [Thread-2] WARN  kafka.consumer.RangeAssignor  - No broker partitions consumed by consumer thread space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b-0 for topic secure_com_from_earth_channel_1
10631 [Thread-2] INFO  kafka.consumer.ZookeeperConsumerConnector  - [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b], Consumer space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b selected partitions : 
10633 [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b-leader-finder-thread] INFO  kafka.consumer.ConsumerFetcherManager$LeaderFinderThread  - [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b-leader-finder-thread], Starting 
10636 [Thread-2] INFO  kafka.consumer.ZookeeperConsumerConnector  - [space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b], end rebalancing consumer space.exploration.mars.rover_Sankets-MacBook-Pro.local-1493604870440-f470846b try #0
10776 [main] INFO  kafka.client.ClientUtils$  - Fetching metadata from broker BrokerEndPoint(0,localhost,9092) with correlation id 0 for 1 topic(s) Set(secure_com_to_earth_channel_1)
10781 [main] INFO  kafka.producer.SyncProducer  - Connected to localhost:9092 for producing
10794 [main] INFO  kafka.producer.SyncProducer  - Disconnecting from localhost:9092
10805 [main] WARN  kafka.producer.BrokerPartitionInfo  - Error while fetching metadata [{TopicMetadata for topic secure_com_to_earth_channel_1 -> 
No partition metadata for topic secure_com_to_earth_channel_1 due to kafka.common.LeaderNotAvailableException}] for topic [secure_com_to_earth_channel_1]: class kafka.common.LeaderNotAvailableException 
10806 [main] INFO  kafka.client.ClientUtils$  - Fetching metadata from broker BrokerEndPoint(0,localhost,9092) with correlation id 1 for 1 topic(s) Set(secure_com_to_earth_channel_1)
10807 [main] INFO  kafka.producer.SyncProducer  - Connected to localhost:9092 for producing
10809 [main] INFO  kafka.producer.SyncProducer  - Disconnecting from localhost:9092
10809 [main] WARN  kafka.producer.BrokerPartitionInfo  - Error while fetching metadata [{TopicMetadata for topic secure_com_to_earth_channel_1 -> 
No partition metadata for topic secure_com_to_earth_channel_1 due to kafka.common.LeaderNotAvailableException}] for topic [secure_com_to_earth_channel_1]: class kafka.common.LeaderNotAvailableException 
10810 [main] ERROR kafka.producer.async.DefaultEventHandler  - Failed to collate messages by topic, partition due to: Failed to fetch topic metadata for topic: secure_com_to_earth_channel_1
10810 [main] INFO  kafka.producer.async.DefaultEventHandler  - Back off for 100 ms before retrying send. Remaining retries = 3
10911 [main] INFO  kafka.client.ClientUtils$  - Fetching metadata from broker BrokerEndPoint(0,localhost,9092) with correlation id 2 for 1 topic(s) Set(secure_com_to_earth_channel_1)
10912 [main] INFO  kafka.producer.SyncProducer  - Connected to localhost:9092 for producing
10914 [main] INFO  kafka.producer.SyncProducer  - Disconnecting from localhost:9092
10914 [main] WARN  kafka.producer.BrokerPartitionInfo  - Error while fetching metadata [{TopicMetadata for topic secure_com_to_earth_channel_1 -> 
No partition metadata for topic secure_com_to_earth_channel_1 due to kafka.common.LeaderNotAvailableException}] for topic [secure_com_to_earth_channel_1]: class kafka.common.LeaderNotAvailableException 
10915 [main] INFO  kafka.client.ClientUtils$  - Fetching metadata from broker BrokerEndPoint(0,localhost,9092) with correlation id 3 for 1 topic(s) Set(secure_com_to_earth_channel_1)
10916 [main] INFO  kafka.producer.SyncProducer  - Connected to localhost:9092 for producing
10917 [main] INFO  kafka.producer.SyncProducer  - Disconnecting from localhost:9092
10918 [main] WARN  kafka.producer.BrokerPartitionInfo  - Error while fetching metadata [{TopicMetadata for topic secure_com_to_earth_channel_1 -> 
No partition metadata for topic secure_com_to_earth_channel_1 due to kafka.common.LeaderNotAvailableException}] for topic [secure_com_to_earth_channel_1]: class kafka.common.LeaderNotAvailableException 
10918 [main] ERROR kafka.producer.async.DefaultEventHandler  - Failed to collate messages by topic, partition due to: Failed to fetch topic metadata for topic: secure_com_to_earth_channel_1
10918 [main] INFO  kafka.producer.async.DefaultEventHandler  - Back off for 100 ms before retrying send. Remaining retries = 2
11020 [main] INFO  kafka.client.ClientUtils$  - Fetching metadata from broker BrokerEndPoint(0,localhost,9092) with correlation id 4 for 1 topic(s) Set(secure_com_to_earth_channel_1)
11021 [main] INFO  kafka.producer.SyncProducer  - Connected to localhost:9092 for producing
11023 [main] INFO  kafka.producer.SyncProducer  - Disconnecting from localhost:9092
11023 [main] WARN  kafka.producer.BrokerPartitionInfo  - Error while fetching metadata [{TopicMetadata for topic secure_com_to_earth_channel_1 -> 
No partition metadata for topic secure_com_to_earth_channel_1 due to kafka.common.LeaderNotAvailableException}] for topic [secure_com_to_earth_channel_1]: class kafka.common.LeaderNotAvailableException 
11024 [main] INFO  kafka.client.ClientUtils$  - Fetching metadata from broker BrokerEndPoint(0,localhost,9092) with correlation id 5 for 1 topic(s) Set(secure_com_to_earth_channel_1)
11024 [main] INFO  kafka.producer.SyncProducer  - Connected to localhost:9092 for producing
11026 [main] INFO  kafka.producer.SyncProducer  - Disconnecting from localhost:9092
11026 [main] WARN  kafka.producer.BrokerPartitionInfo  - Error while fetching metadata [{TopicMetadata for topic secure_com_to_earth_channel_1 -> 
No partition metadata for topic secure_com_to_earth_channel_1 due to kafka.common.LeaderNotAvailableException}] for topic [secure_com_to_earth_channel_1]: class kafka.common.LeaderNotAvailableException 
11027 [main] ERROR kafka.producer.async.DefaultEventHandler  - Failed to collate messages by topic, partition due to: Failed to fetch topic metadata for topic: secure_com_to_earth_channel_1
11027 [main] INFO  kafka.producer.async.DefaultEventHandler  - Back off for 100 ms before retrying send. Remaining retries = 1
11128 [main] INFO  kafka.client.ClientUtils$  - Fetching metadata from broker BrokerEndPoint(0,localhost,9092) with correlation id 6 for 1 topic(s) Set(secure_com_to_earth_channel_1)
11129 [main] INFO  kafka.producer.SyncProducer  - Connected to localhost:9092 for producing
11131 [main] INFO  kafka.producer.SyncProducer  - Disconnecting from localhost:9092
11131 [main] WARN  kafka.producer.BrokerPartitionInfo  - Error while fetching metadata [{TopicMetadata for topic secure_com_to_earth_channel_1 -> 
No partition metadata for topic secure_com_to_earth_channel_1 due to kafka.common.LeaderNotAvailableException}] for topic [secure_com_to_earth_channel_1]: class kafka.common.LeaderNotAvailableException 
11131 [main] INFO  kafka.client.ClientUtils$  - Fetching metadata from broker BrokerEndPoint(0,localhost,9092) with correlation id 7 for 1 topic(s) Set(secure_com_to_earth_channel_1)
11132 [main] INFO  kafka.producer.SyncProducer  - Connected to localhost:9092 for producing
11133 [main] INFO  kafka.producer.SyncProducer  - Disconnecting from localhost:9092
11136 [main] WARN  kafka.producer.BrokerPartitionInfo  - Error while fetching metadata [{TopicMetadata for topic secure_com_to_earth_channel_1 -> 
No partition metadata for topic secure_com_to_earth_channel_1 due to kafka.common.LeaderNotAvailableException}] for topic [secure_com_to_earth_channel_1]: class kafka.common.LeaderNotAvailableException 
11136 [main] ERROR kafka.producer.async.DefaultEventHandler  - Failed to collate messages by topic, partition due to: Failed to fetch topic metadata for topic: secure_com_to_earth_channel_1
11136 [main] INFO  kafka.producer.async.DefaultEventHandler  - Back off for 100 ms before retrying send. Remaining retries = 0
11237 [main] INFO  kafka.client.ClientUtils$  - Fetching metadata from broker BrokerEndPoint(0,localhost,9092) with correlation id 8 for 1 topic(s) Set(secure_com_to_earth_channel_1)
11238 [main] INFO  kafka.producer.SyncProducer  - Connected to localhost:9092 for producing
11240 [main] INFO  kafka.producer.SyncProducer  - Disconnecting from localhost:9092
11240 [main] WARN  kafka.producer.BrokerPartitionInfo  - Error while fetching metadata [{TopicMetadata for topic secure_com_to_earth_channel_1 -> 
No partition metadata for topic secure_com_to_earth_channel_1 due to kafka.common.LeaderNotAvailableException}] for topic [secure_com_to_earth_channel_1]: class kafka.common.LeaderNotAvailableException 
11241 [main] ERROR kafka.producer.async.DefaultEventHandler  - Failed to send requests for topics secure_com_to_earth_channel_1 with correlation ids in [0,8]
33264 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Unable to read additional data from server sessionid 0x15b9b2525a100d4, likely server has closed socket, closing socket connection and attempting reconnect
33368 [main-EventThread] INFO  org.I0Itec.zkclient.ZkClient  - zookeeper state changed (Disconnected)
34349 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Opening socket connection to server localhost/0:0:0:0:0:0:0:1:2181. Will not attempt to authenticate using SASL (unknown error)
34350 [main-SendThread(localhost:2181)] WARN  org.apache.zookeeper.ClientCnxn  - Session 0x15b9b2525a100d4 for server null, unexpected error, closing socket connection and attempting reconnect
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1081)
36448 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error)
36449 [main-SendThread(localhost:2181)] WARN  org.apache.zookeeper.ClientCnxn  - Session 0x15b9b2525a100d4 for server null, unexpected error, closing socket connection and attempting reconnect
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1081)
37201 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Opening socket connection to server localhost/0:0:0:0:0:0:0:1:2181. Will not attempt to authenticate using SASL (unknown error)
37201 [main-SendThread(localhost:2181)] WARN  org.apache.zookeeper.ClientCnxn  - Session 0x15b9b2525a100d4 for server null, unexpected error, closing socket connection and attempting reconnect
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1081)
38689 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error)
38689 [main-SendThread(localhost:2181)] WARN  org.apache.zookeeper.ClientCnxn  - Session 0x15b9b2525a100d4 for server null, unexpected error, closing socket connection and attempting reconnect
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1081)
39077 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Opening socket connection to server localhost/0:0:0:0:0:0:0:1:2181. Will not attempt to authenticate using SASL (unknown error)
39077 [main-SendThread(localhost:2181)] WARN  org.apache.zookeeper.ClientCnxn  - Session 0x15b9b2525a100d4 for server null, unexpected error, closing socket connection and attempting reconnect
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1081)
41049 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error)
41049 [main-SendThread(localhost:2181)] WARN  org.apache.zookeeper.ClientCnxn  - Session 0x15b9b2525a100d4 for server null, unexpected error, closing socket connection and attempting reconnect
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1081)
41645 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Opening socket connection to server localhost/0:0:0:0:0:0:0:1:2181. Will not attempt to authenticate using SASL (unknown error)
41646 [main-SendThread(localhost:2181)] WARN  org.apache.zookeeper.ClientCnxn  - Session 0x15b9b2525a100d4 for server null, unexpected error, closing socket connection and attempting reconnect
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1081)
42955 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error)
42956 [main-SendThread(localhost:2181)] WARN  org.apache.zookeeper.ClientCnxn  - Session 0x15b9b2525a100d4 for server null, unexpected error, closing socket connection and attempting reconnect
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1081)
43251 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Opening socket connection to server localhost/0:0:0:0:0:0:0:1:2181. Will not attempt to authenticate using SASL (unknown error)
43252 [main-SendThread(localhost:2181)] WARN  org.apache.zookeeper.ClientCnxn  - Session 0x15b9b2525a100d4 for server null, unexpected error, closing socket connection and attempting reconnect
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1081)
44761 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error)
44762 [main-SendThread(localhost:2181)] WARN  org.apache.zookeeper.ClientCnxn  - Session 0x15b9b2525a100d4 for server null, unexpected error, closing socket connection and attempting reconnect
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1081)
45757 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Opening socket connection to server localhost/0:0:0:0:0:0:0:1:2181. Will not attempt to authenticate using SASL (unknown error)
45758 [main-SendThread(localhost:2181)] WARN  org.apache.zookeeper.ClientCnxn  - Session 0x15b9b2525a100d4 for server null, unexpected error, closing socket connection and attempting reconnect
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:361)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1081)
47585 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Opening socket connection to server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error)
47586 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Socket connection established to localhost/127.0.0.1:2181, initiating session
47588 [main-SendThread(localhost:2181)] INFO  org.apache.zookeeper.ClientCnxn  - Session establishment complete on server localhost/127.0.0.1:2181, sessionid = 0x15b9b2525a100d4, negotiated timeout = 6000
47588 [main-EventThread] INFO  org.I0Itec.zkclient.ZkClient  - zookeeper state changed (SyncConnected)
